qiskit_nature/algorithms/ground_state_solvers/adapt_vqe.py

```diff
@@ -58,23 +81,43 @@ def __init__(
             delta: the finite difference step size for the gradient computation. It has a minimum
                 value of 1e-5.
             max_iterations: the maximum number of iterations of the AdaptVQE algorithm.
+            gradient: a class that converts operator expression to the first-order gradient based
+                on the method mentioned.
         """
         validate_min("threshold", threshold, 1e-15)
-        validate_min("delta", delta, 1e-5)
+
+        if isinstance(gradient, float):
+            # this scenario can only occur while using the deprecate_arguments wrapper which will
+            # move any argument supplied to delta into gradient.
+            gradient = Gradient(grad_method="fin_diff", epsilon=gradient)
 
         super().__init__(qubit_converter, solver)
 
         self._threshold = threshold
-        self._delta = delta
         self._max_iterations = max_iterations
+        self.gradient = gradient  # use the property setter to handle `None` case consistently
 
         self._excitation_pool: List[OperatorBase] = []
         self._excitation_list: List[OperatorBase] = []
 
         self._main_operator: PauliSumOp = None
         self._ansatz: QuantumCircuit = None
+        self._sampler: CircuitSampler = None
+
+    @property
+    def gradient(self) -> Optional[GradientBase]:
```

Comment:
Now this cannot be `None` anymore, right? 🙂 
```suggestion
    def gradient(self) -> GradientBase:
```